### PR TITLE
Slider board midi

### PR DESCRIPTION
**Pull request by mithrandir123**
- Added support for Behringer BCF2000 midi slider board.
 - Hooked up the slider board to joint teleop panel.

When the board is not connect, joint teleop behaves as usual.

https://youtu.be/Nwju1w1DaPM

### CODE REVIEW

File: src/python/director/teleoppanel.py [hunk 1]
```diff
@@ -1107,6 +1154,27 @@ def buildTabWidget(self, jointGroups):
 
         self.signalMapper.connect('mapped(const QString&)', self.sliderChanged)
 
+    def tabChenged(self, id):
```
Comment (iamwolf) [inline src/python/director/teleoppanel.py]: *chAnged

**Comment by patmarion** [comment]
Hey this is great!  I want to get one of those!

I think this is awesome, but I don't want to merge this change as is.  I think it needs to be designed so that the midi is decoupled from the UI.  The panel class shouldn't have a behringer member variable, and it shouldn't have a member function named setupSliderBoard().  Likewise, a function like slider board that deals with midi values shouldn't know anything about joint names.  Instead, maybe we could design a very general class that can manage the sync between a Qt slider and a midi device.  Next, the teleop panel class can be extended to provide whatever general purpose API is required to attach a specific instance of this manager object which handles the mapping of joint names to sliders.  I know this isn't a very clear description, but something like that.  Ideally, the only changes to teleoppanel.py are just changes that add generic api.

**Comment by mithrandir123** [comment]
I was afraid you were going to say exactly that :)
Alas, I agree.

**Comment by patmarion** [comment]
I'm going to close this for now.  But I hope the work doesn't disappear, I think it's really great, just needs to be moved into a standalone module and then it will be easier to integrate.

**Comment by mithrandir123** [comment]
It's still on my list. I'll reopen the pull request when I get to it.

**Pull request closed without merging**